src/main/java/sonarquberepair/processor/SelfAssignementProcessor.java

```diff
@@ -0,0 +1,86 @@
+package sonarquberepair.processor;
+
+import org.sonar.java.checks.SelfAssignementCheck;
+
+import spoon.reflect.factory.Factory;
+
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.code.CtAssignment;
+import spoon.reflect.code.CtThisAccess;
+import spoon.reflect.code.CtFieldRead;
+import spoon.reflect.code.CtFieldAccess;
+import spoon.reflect.code.CtArrayAccess;
+import spoon.reflect.code.CtVariableAccess;
+import spoon.reflect.code.CtTargetedExpression;
+import spoon.reflect.declaration.CtField;
+import spoon.reflect.declaration.CtVariable;
+import spoon.reflect.declaration.CtType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class SelfAssignementProcessor extends SQRAbstractProcessor<CtAssignment<?,?>> {
+
+	public SelfAssignementProcessor(String originalFilesPath) {
+		super(originalFilesPath, new SelfAssignementCheck());
+	}
+
+	@Override
+	public boolean isToBeProcessed(CtAssignment<?,?> candidate) {
+		if (!super.isToBeProcessedAccordingToSonar(candidate)) {
+			return false;
+		}
+
+		CtExpression<?> leftExpression = candidate.getAssigned();
+		CtExpression<?> rightExpression = candidate.getAssignment();
+		if (rightExpression == null || candidate.getParent(CtAssignment.class) != null) {
+			/* Ignore multiple assigment case*/
+			return false;
+		}
+
+		if (leftExpression.toString().equals(rightExpression.toString())) {
+			if (!candidate.getAssigned().toString().contains("this.") && candidate.getAssigned().toString().contains(".")) {
```

Comment:
Why do you need the two checks in lines 42 and 43?

The checking with sonarjava should be already enough, no?

Reply:
The problem is sonarjava will also return true for cases such as `a.b = a.b` is not handled by this processor, therefore I only use it as a soft-check then hard check with spoon to filter out cases this processor can handle.

Reply:
- Why is there a case that is not handled by this processor? We need to document this.

- The case you just mentioned, `a.b = a.b`, is not related to the check in line 42. The check in line 42 will always be true if it's executed, because sonarjava will not return true if the expressions are not the same.

Reply:
The motivation is the case `a.b = a.b` is very uncommon (unless you delibrately want to do it) and since it requires to check the class file to look for `CtField` , it might not be in classpath. The most common cases would always occur in the same class file such as `a = a` .

Reply:
> the case a.b = a.b is very uncommon

I don't know where you got this information from, but it's your call.

What about the second point I mentioned?

Reply:
Right that is fixed :) .

Reply:
@fermadeiral I just checked apparently sonarjava also return true even when the expression does not look the same . I need both left and right to look the same to fix in this case :) . therefore line 42 need to kept.

Reply:
The new updated processor now also handle `a.b = a.b` or `a.getSelf().b = a.getSelf().b` by deleting it once it's seen.

Reply:
> I just checked apparently sonarjava also return true even when the expression does not look the same, for instance this.a = a regarded as a false negative in their test file.

That is pretty bad, thank you for letting me know.

> The new updated processor now also handle a.b = a.b or a.getSelf().b = a.getSelf().b by deleting it once it's seen.

Perfect. I will review the entire changes now.

Reply:
no problem . There were so many details regarding this particular rules. I am glad you asked. >(